src/qaqc/01-organize_raw_data.Rmd

```diff
@@ -20,83 +20,96 @@ all_data <- list.files(path = 'data/api/', full.names = TRUE) %>%
+                      
+                      #Dissolved Oxygen
+                      rdo_slope = str_match(rdo, "Slope\\s*(.*?)\\s*Offset")[,2],
+                      rdo_offset = str_match(rdo, "Offset\\s*(.*?)\\s*mg/L")[,2],
+                      rdo_100 = str_match(rdo, "PreMeasurement\\s*(.*?)\\s*%SatPost")[,2],
+                      rdo_conc = str_match(rdo, "Concentration\\s*(.*?)\\s*mg/LPreMeasurement")[,2],
+                      rdo_temp = str_match(rdo, "Temperature\\s*(.*?)\\s*°C")[,2],
+                      rdo_pressure = str_match(rdo, "Pressure\\s*(.*?)\\s*mbar")[,2],
+                      
+                      #pH
+                      ph_slope_pre = str_match(ph_orp, "Offset1Slope\\s*(.*?)\\s*mV/pH")[,2],
+                      ph_offset_pre = str_match(ph_orp, "mV/pHOffset\\s*(.*?)\\s*mVSlopeandOffset2")[,2],
+                      ph_slope_post = str_match(ph_orp, "Offset2Slope\\s*(.*?)\\s*mV/pH")[,2],
+                      ph_offset_post = str_match(ph_orp, paste0(ph_slope_post,"mV/pHOffset\\s*(.*?)\\s*mVORPORP"))[,2],
+                      ph_7_nice = str_sub(str_match(ph_orp, "PostMeasurementpH7\\s*(.*?)\\s*mVCal")[,2], 10, nchar(str_match(ph_orp, "PostMeasurementpH7\\s*(.*?)\\s*mVCal")[,2])),
+                      ph_7_other = str_sub(str_match(ph_orp, "PostMeasurementpH6\\s*(.*?)\\s*mVCal")[,2], 10, nchar(str_match(ph_orp, "PostMeasurementpH6\\s*(.*?)\\s*mVCal")[,2])),
+                      ph_7 = ifelse(is.na(ph_7_nice), ph_7_other, ph_7_nice),
+                      
+                      #ORP
+                      orp_offset = ifelse(is.na(str_match(ph_orp, "Zobell'sOffset\\s*(.*?)\\s*mVTemperature")[,2]),
+                                          str_match(ph_orp, "ZoBell'sOffset\\s*(.*?)\\s*mVTemperature")[,2],
+                                          str_match(ph_orp, "Zobell'sOffset\\s*(.*?)\\s*mVTemperature")[,2]),
+                      
+                      #Conductivity
+                      tds_conversion_ppm = str_sub(str_match(conductivity, "TDSConversionFactor\\s*(.*?)\\s*CellConstant")[,2], 6, nchar(str_match(conductivity, "TDSConversionFactor\\s*(.*?)\\s*CellConstant")[,2])),
+                      cond_cell_constant = str_match(conductivity, "CellConstant\\s*(.*?)\\s*ReferenceTemperature")[,2],
+                      cond_pre = str_match(conductivity,paste0(str_match(conductivity,
+                                                                         "PreMeasurementActual\\s*(.*?)\\s*SpecificConductivity")[,2],"SpecificConductivity\\s*(.*?)\\s*µS/cmPost"))[,2],
+                      cond_post = str_match(conductivity,paste0(str_match(conductivity,
+                                                                          "PostMeasurementActual\\s*(.*?)\\s*SpecificConductivity")[,2],"SpecificConductivity\\s*(.*?)\\s*µS/cm"))[,2],
+                      
+                      #Turbidity
+                      ntu_slope = "None",
```

Comment:
Consider using 'NA_character_' here. That way when you coerce all into numeric, it won't give you silly warnings about coercion.